hudi/src/main/scala/io/qbeast/spark/hudi/HudiMetadataManager.scala

```diff
@@ -0,0 +1,166 @@
+import io.qbeast.IISeq
+import org.apache.hadoop.fs.FileSystem
+import org.apache.hadoop.fs.Path
+import org.apache.hudi.common.config.TypedProperties
+import org.apache.hudi.common.model.HoodieFileFormat
+import org.apache.hudi.common.model.HoodieTableType
+import org.apache.hudi.common.model.HoodieTimelineTimeZone
+import org.apache.hudi.common.table.HoodieTableMetaClient
+import org.apache.hudi.config.HoodieWriteConfig
+import org.apache.hudi.hadoop.fs.HadoopFSUtils
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.SaveMode
+import org.apache.spark.sql.SparkSession
+
+import java.nio.file.Paths
+import scala.jdk.CollectionConverters.mapAsJavaMapConverter
+
+/**
+ * Spark+Hudi implementation of the MetadataManager interface
+ */
+object HudiMetadataManager extends MetadataManager {
+
+  private val spark = SparkSession.active
+  private val hadoopConf = spark.sparkContext.hadoopConfiguration
+
+  override def updateWithTransaction(
+      tableID: QTableID,
+      schema: StructType,
+      options: QbeastOptions,
+      writeMode: WriteModeValue)(
+      writer: String => (TableChanges, IISeq[IndexFile], IISeq[DeleteFile])): Unit = {
+
+    // Hudi requires the table name to be set. For now in here we obtain the table name from the path
+    val extraOptions =
+      if (!options.extraOptions.contains(HoodieWriteConfig.TBL_NAME.key))
+        options.extraOptions ++ Map(
+          HoodieWriteConfig.TBL_NAME.key -> Paths.get(tableID.id).getFileName.toString)
+      else options.extraOptions
+    val newOptions = options.copy(extraOptions = extraOptions)
+
+    if (!existsLog(tableID)) createTable(tableID, newOptions.extraOptions)
+
+    val metaClient = loadMetaClient(tableID)
+    val metadataWriter =
+      HudiMetadataWriter(tableID, writeMode, metaClient, newOptions, schema)
+
+    metadataWriter.writeWithTransaction(writer)
+  }
+
+  override def updateMetadataWithTransaction(
+      tableID: QTableID,
+      dataSchema: StructType,
+      overwrite: Boolean)(config: => Configuration): Unit = {
+
+    val metaClient = loadMetaClient(tableID)
+    val metadataWriter =
+      HudiMetadataWriter(
+        tableID,
+        SaveMode.Append.toString,
+        metaClient,
+        QbeastOptions.empty,
+        dataSchema)
+
+    metadataWriter.updateMetadataWithTransaction(config, overwrite)
+
+  }
+
+  override def loadSnapshot(tableID: QTableID): HudiQbeastSnapshot = {
+    HudiQbeastSnapshot(tableID)
+  }
+
+  override def loadCurrentSchema(tableID: QTableID): StructType = {
+    loadSnapshot(tableID).schema
+  }
+
+  override def updateRevision(tableID: QTableID, revisionChange: RevisionChange): Unit = {}
+
+  override def updateTable(tableID: QTableID, tableChanges: TableChanges): Unit = {}
+
+  /**
+   * Returns the MetaClient for the table
+   * @param tableID
+   *   the table ID
+   * @return
+   */
+  private def loadMetaClient(tableID: QTableID): HoodieTableMetaClient = {
+    HoodieTableMetaClient
```

Comment:
For now, I'm following the same style as in Delta, where the `loadDeltaLog()` (the equivalent method to this one) is loaded all the time. Probably, it would be convenient to evaluate the potential overhead of this.